2023-09-18 18:06:44.300752: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 18:06:44.300935: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 18:06:51,878 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 18:06:51,942 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 18:06:51,943 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 18:06:51,943 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 18:06:52,913 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 18:06:52,913 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:52,928 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.43333333333333335}
INFO flwr 2023-09-18 18:06:52,929 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 18:06:53,122 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 18:06:53,145 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 18:06:53,147 | fedavgm.py:214 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,155 | server.py:125 | fit progress: (1, 0.3266284859084374, {'accuracy': 0.8533333333333334}, 0.22591440000000063)
DEBUG flwr 2023-09-18 18:06:53,155 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 18:06:53,175 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 18:06:53,176 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 18:06:53,176 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 18:06:53,201 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,211 | server.py:125 | fit progress: (2, 0.32575052033541596, {'accuracy': 0.8566666666666667}, 0.28173980000000043)
DEBUG flwr 2023-09-18 18:06:53,211 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 18:06:53,230 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,230 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 18:06:53,251 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,261 | server.py:125 | fit progress: (3, 0.32574754918328525, {'accuracy': 0.8566666666666667}, 0.33248980000000117)
DEBUG flwr 2023-09-18 18:06:53,261 | server.py:173 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 18:06:53,284 | server.py:187 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,284 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 18:06:53,312 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,324 | server.py:125 | fit progress: (4, 0.3294013689326852, {'accuracy': 0.8566666666666667}, 0.39555209999999974)
DEBUG flwr 2023-09-18 18:06:53,324 | server.py:173 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 18:06:53,346 | server.py:187 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,347 | server.py:223 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 18:06:53,375 | server.py:237 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,385 | server.py:125 | fit progress: (5, 0.32940416131305755, {'accuracy': 0.8566666666666667}, 0.4565987000000007)
DEBUG flwr 2023-09-18 18:06:53,386 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 18:06:53,410 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,410 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 18:06:53,442 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,453 | server.py:125 | fit progress: (6, 0.33188964495098505, {'accuracy': 0.8566666666666667}, 0.5239375000000006)
DEBUG flwr 2023-09-18 18:06:53,453 | server.py:173 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 18:06:53,479 | server.py:187 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,479 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,517 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,529 | server.py:125 | fit progress: (7, 0.3307980740076461, {'accuracy': 0.8533333333333334}, 0.6005205)
DEBUG flwr 2023-09-18 18:06:53,530 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,554 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,554 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,585 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,598 | server.py:125 | fit progress: (8, 0.33078792358080583, {'accuracy': 0.8533333333333334}, 0.6690217999999994)
DEBUG flwr 2023-09-18 18:06:53,598 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,622 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,623 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,656 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,669 | server.py:125 | fit progress: (9, 0.33078870201740734, {'accuracy': 0.8533333333333334}, 0.7408304999999995)
DEBUG flwr 2023-09-18 18:06:53,669 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,696 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 18:06:53,696 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,728 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 18:06:53,739 | server.py:125 | fit progress: (10, 0.33078864067055497, {'accuracy': 0.8533333333333334}, 0.8106983000000003)
DEBUG flwr 2023-09-18 18:06:53,740 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 18:06:53,764 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 18:06:53,764 | server.py:147 | FL finished in 0.8356393999999998
INFO flwr 2023-09-18 18:06:53,765 | app.py:218 | app_fit: losses_distributed [(1, 0.3543968088924885), (2, 0.3543000463396311), (3, 0.3762493186763355), (4, 0.3753137098891394), (5, 0.38383060480867115), (6, 0.38306050925027757), (7, 0.3879264175891876), (8, 0.3879267726625715), (9, 0.3879268773964473), (10, 0.3879268548318318)]
INFO flwr 2023-09-18 18:06:53,765 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 18:06:53,765 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 18:06:53,765 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.3266284859084374), (2, 0.32575052033541596), (3, 0.32574754918328525), (4, 0.3294013689326852), (5, 0.32940416131305755), (6, 0.33188964495098505), (7, 0.3307980740076461), (8, 0.33078792358080583), (9, 0.33078870201740734), (10, 0.33078864067055497)]
INFO flwr 2023-09-18 18:06:53,765 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.43333333333333335), (1, 0.8533333333333334), (2, 0.8566666666666667), (3, 0.8566666666666667), (4, 0.8566666666666667), (5, 0.8566666666666667), (6, 0.8566666666666667), (7, 0.8533333333333334), (8, 0.8533333333333334), (9, 0.8533333333333334), (10, 0.8533333333333334)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[-2.27912388,  1.23408997,  0.89899138,  0.4595263 ,  0.24048244,
         0.07484587,  0.03695987, -0.01609832]]), array([0.0076575])]
[array([[-2.26841229,  1.24279797,  0.92187701,  0.46139717,  0.21623002,
         0.06449821,  0.0460821 , -0.02019132]]), array([-0.0160538])]
[array([[-2.26864262,  1.24292437,  0.9218433 ,  0.46163384,  0.21617793,
         0.0645277 ,  0.04617928, -0.02035198]]), array([-0.01605841])]
[array([[-2.18186236,  1.1196954 ,  0.82936577,  0.44915568,  0.25494179,
         0.03460116,  0.08954354, -0.01589148]]), array([0.03940521])]
[array([[-2.18183784,  1.11960638,  0.82938566,  0.44895136,  0.25500037,
         0.03462354,  0.08950051, -0.01583123]]), array([0.03956431])]
[array([[-2.15592716,  1.05108154,  0.82019594,  0.4258054 ,  0.26860978,
         0.0563778 , -0.0108917 , -0.03647643]]), array([0.17208098])]
[array([[-2.12606325,  1.03829599,  0.85290473,  0.43758646,  0.23373709,
         0.06496701,  0.00585885, -0.02984901]]), array([0.05813064])]
[array([[-2.12563015,  1.03858939,  0.85312819,  0.43766584,  0.23388983,
         0.06514941,  0.00623673, -0.02959278]]), array([0.05585667])]
[array([[-2.12564184,  1.03860842,  0.85305558,  0.43763224,  0.23383992,
         0.065203  ,  0.00616914, -0.02961193]]), array([0.05606056])]
[array([[-2.12564966,  1.03861512,  0.85305614,  0.43762822,  0.2338537 ,
         0.06519814,  0.00617519, -0.02961246]]), array([0.05605231])]
>>>>>>>>>>>>>run time:
0.5625s
